api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java

```diff
@@ -813,6 +813,321 @@ public Long getCountOfPatients(String query, boolean includeVoided) {
+		if (tmpStart == null) {
+			tmpStart = 0;
+		}
+		Integer maxLength = HibernatePersonDAO.getMaximumSearchResults();
+		Integer tmpLength = length;
+		if (tmpLength == null || tmpLength > maxLength) {
+			tmpLength = maxLength;
+		}
+
+		List<Patient> patients = new LinkedList<>();
+
+		PatientLuceneQuery personLuceneQuery = new PatientLuceneQuery(sessionFactory);
+		LuceneQuery<Person> ageQuery = personLuceneQuery.getPatinetWithAgeRange(from, to, includeVoided);
+		
+		long namesSize = ageQuery.resultSize();
+		if (namesSize > tmpStart) {
+			ListPart<Object[]> patientAge = ageQuery.listPartProjection(tmpStart, tmpLength, "personId");
+			patientAge.getList().forEach(patient -> patients.add(getPatient((Integer) patient[0])));
+		} 
+
+		return patients;
+	}
+	
+	@Override
+	public List<Patient> getPatients(Date birthdate, Integer start, Integer length, Boolean includeVoided)
+			throws DAOException {
+		Integer tmpStart = start;
+		if (tmpStart == null) {
+			tmpStart = 0;
+		}
+		Integer maxLength = HibernatePersonDAO.getMaximumSearchResults();
+		Integer tmpLength = length;
+		if (tmpLength == null || tmpLength > maxLength) {
+			tmpLength = maxLength;
+		}
+
+		List<Patient> patients = new LinkedList<>();
+
+		
+		
+		PatientLuceneQuery personLuceneQuery = new PatientLuceneQuery(sessionFactory);
+		LuceneQuery<Person> birthdateQuery = personLuceneQuery.getPatinetWithBirthdate(birthdate, includeVoided);
+		
+		long namesSize = birthdateQuery.resultSize();
+		if (namesSize > tmpStart) {
+			ListPart<Object[]> patientBirthdate = birthdateQuery.listPartProjection(tmpStart, tmpLength, "personId");
+			patientBirthdate.getList().forEach(patient -> patients.add(getPatient((Integer) patient[0])));
+		} 
+
+		return patients;
+	}
+	
+	@Override
+	public List<Patient> getPatients(String query, String gender, Integer start, Integer length, Boolean includeVoided)
+	        throws DAOException {
+		List<Patient> patients = findPatients(query, includeVoided, start, length);
+		List<Patient> result=new LinkedList<>();
+		for (Patient p : patients) {
+			if (p.getGender().equals(gender)) {
+				result.add(p);
+			}
+		}
+		return result;
+	}
+	
+	@Override
+	public List<Patient> getPatients(String query, Date from, Date to, Integer start, Integer length,
+	        Boolean includeVoided) throws DAOException {
+		Integer tmpStart = start;
+		if (tmpStart == null) {
+			tmpStart = 0;
+		}
+		Integer maxLength = HibernatePersonDAO.getMaximumSearchResults();
+		Integer tmpLength = length;
+		if (tmpLength == null || tmpLength > maxLength) {
+			tmpLength = maxLength;
+		}
+		query = LuceneQuery.escapeQuery(query);
+
+		List<Patient> patients = new LinkedList<>();
+
+		String minChars = Context.getAdministrationService().getGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_MIN_SEARCH_CHARACTERS);
+
+		if (minChars == null || !StringUtils.isNumeric(minChars)) {
+			minChars = "" + OpenmrsConstants.GLOBAL_PROPERTY_DEFAULT_MIN_SEARCH_CHARACTERS;
```

Comment:
any reason to append "" in the beginning ?

Reply:
For casting int to string, although I removed that part